integration/restore_test.go

```diff
@@ -0,0 +1,123 @@
+// +build integration
+
+package integration
```

Comment:
I would actually move this test closer to restore package as well and name it `restore_integration_test.go`.
To me, integration tests feel more closely to unit tests than e2e tests. The only difference is the API server

Reply:
I'll move them to the controller package then, since that's the entrypoint to the reconciliation.